lib/vhx.js

```diff
@@ -1,61 +1,55 @@
-import defaults from './defaults';
+'use strict';
+
+import ApiConfig from './config';
 import Collection from './resources/collections';
 import Video from './resources/videos';
 import Customer from './resources/customers';
-import Product from './resources/products';
 import Browse from './resources/browse';
-import request from 'superagent';
+import Product from './resources/products';
+import Analytics from './resources/analytics';
 
-let resources = {
-  collections : Collection,
-  videos      : Video,
+const resources = {
   products    : Product,
+  browse      : Browse,
+  videos      : Video,
+  collections : Collection,
   customers   : Customer,
-  browse      : Browse
+  analytics   : Analytics
 };
 
-class vhx {
+class VhxApi {
   constructor(key, opts) {
-    let _this = this;
-
-    if (!(_this instanceof vhx)) {
-      return new vhx(key);
-    }
+    this.api = opts.internal ? this.setToken(key, opts) : this.setApi(key, opts);
 
-    _this.api = opts.token_host ? _this.setToken(key, opts) : _this.setApi(key, opts);
-
-    _this.prepareResources();
+    this.prepareResources();
   }
 
   setApi(key, opts = {}) {
     return {
       auth: 'Basic ' + btoa(key),
-      host: opts.host || defaults.HOST,
-      protocol: opts.protocol || defaults.PROTOCOL,
-      timeout: defaults.TIMEOUT,
+      host: opts.host || ApiConfig.HOST,
+      protocol: opts.protocol || ApiConfig.PROTOCOL,
+      timeout: ApiConfig.TIMEOUT,
       token_expiration: null
     };
   }
 
   setToken(key, opts = {}) {
     return {
       token: key,
-      host: opts.host || defaults.HOST,
-      protocol: opts.protocol || defaults.PROTOCOL,
-      timeout: defaults.TIMEOUT,
-      token_host: opts.token_host || defaults.TOKEN_HOST,
-      token_expiration: TOKEN_EXPIRES_IN
+      host: opts.host || ApiConfig.HOST,
+      protocol: opts.protocol || ApiConfig.PROTOCOL,
+      timeout: ApiConfig.TIMEOUT,
+      token_expiration: TOKEN_EXPIRES_IN,
+      internal: true
     };
   }
 
   prepareResources() {
-    let _this = this;
-
     for (let name in resources) {
```

Comment:
Since I see a lot of changes from `let` to `const` where appropriate (which is a good idea!), this is one more place it can be swapped out. Use `for (const name in resources) { ... }`.

Seems strange since `name` is reassigned on every iteration of the loop, but as long as it's not assigned to anything within the loop it's valid syntax.